app/src/main/java/fr/neamar/kiss/result/AppResult.java

```diff
@@ -109,11 +110,31 @@ protected Boolean popupMenuClickHandler(Context context, RecordAdapter parent, M
             case R.id.item_app_hibernate:
                 hibernate(context, appPojo);
                 return true;
+            case R.id.item_exclude:
+                // remove item since it will be hiddden
+                parent.removeResult(this);
+                excludeFromSettings(context, appPojo);
+                return true;
         }
 
         return super.popupMenuClickHandler(context, parent, item);
     }
 
+    private void excludeFromSettings(Context context, AppPojo appPojo)
+    {
+
+        String excludedAppList = PreferenceManager.getDefaultSharedPreferences(context).
+                getString("excluded-apps-list", context.getPackageName() + ";");
+        PreferenceManager.getDefaultSharedPreferences(context).edit()
+                .putString("excluded-apps-list", excludedAppList + appPojo.packageName + ";").commit();
+        //remove app pojo from appProvider results - no need to reset handler
+        KissApplication.getDataHandler(context).getAppProvider().removeApp(appPojo);
+        if(context instanceof MainActivity) {
+            ((MainActivity) context).retrieveFavorites();
```

Comment:
Yes, exactly! I renamed the method to excludeFromAppList since the previous name didnt say much. 
And maybe retrieveFavorites should be reloadFavorites (I leave this to you)